headerCheck/factory/nodesCoordinator.go

```diff
@@ -0,0 +1,132 @@
+package factory
+
+import (
+	"fmt"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/data/endProcess"
+	"github.com/ElrondNetwork/elrond-go-core/hashing"
+	"github.com/ElrondNetwork/elrond-go-core/hashing/sha256"
+	"github.com/ElrondNetwork/elrond-go-core/marshal"
+	"github.com/ElrondNetwork/elrond-go/config"
+	"github.com/ElrondNetwork/elrond-go/sharding/nodesCoordinator"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/data"
+	"github.com/ElrondNetwork/elrond-sdk-erdgo/disabled"
+)
+
+type validator = nodesCoordinator.Validator
+
+// CreateNodesCoordinator creates nodes coordinator which will be used for header verification
+func CreateNodesCoordinator(
+	hasher hashing.Hasher,
+	marshaller marshal.Marshalizer,
+	rater nodesCoordinator.ChanceComputer,
+	networkConfig *data.NetworkConfig,
+	enableEpochsConfig *data.EnableEpochsConfig,
+) (nodesCoordinator.EpochsConfigUpdateHandler, error) {
+
+	waitingMap := make(map[uint32][]validator)
+	eligibleMap := createDummyNodesMap(networkConfig.MetaConsensusGroup, networkConfig.NumShardsWithoutMeta)
+
+	argsNodesShuffler := createArgsNodesShuffler(enableEpochsConfig, networkConfig)
+	nodeShuffler, err := nodesCoordinator.NewHashValidatorsShuffler(argsNodesShuffler)
+	if err != nil {
+		return nil, err
+	}
+
+	initialEpoch := uint32(0)
+	dummySelfPublicKey := []byte("dummy")
```

Comment:
this should have been the crypto components' publicKey in order to be consistent among the components & their state used

Reply:
changed to use crypto components publicKey